dockers/docker-dhcp-relay/start.sh

```diff
@@ -6,13 +6,16 @@ rm -f /var/run/rsyslogd.pid
 # Start rsyslog
 supervisorctl start rsyslogd
 
-# Wait for all interfaces to come up and be assigned IPv4 addresses before
-# starting the DHCP relay agent(s). If an interface the relay should listen
-# on is down, the relay agent will not start. If an interface the relay should
-# listen on is up but does not have an IP address assigned when the relay
-# agent starts, it will not listen or send on that interface for the lifetime
-# of the process.
-/usr/bin/wait_for_intf.sh
+# If our supervisor config has entries in the "isc-dhcp-relay" group...
+if [ $(supervisorctl status | grep -c "^isc-dhcp-relay:") -gt 0 ]; then
```

Comment:
`if supervisorctl status | grep -q "^isc-dhcp-relay:"; then` do the same, but faster

Reply:
Hmm... "grep -q" actually seems slower for me...

```
root@sonic:/# time supervisorctl status | grep -c "^isc-dhcp-relay:"
1

real    0m0.443s
user    0m0.348s
sys     0m0.088s
root@sonic:/# time supervisorctl status | grep -q "^isc-dhcp-relay:"

real    0m0.470s
user    0m0.384s
sys     0m0.068s
```

Reply:
:) when you write `[` you run an external command in the bash script 
```
$ which [
/usr/bin/[
```
So bash must to run subprocess to calculate the boolean for your `if`
Executing the subprocess is not so fast.
Faster would be use `[[` which is internal bash instruction.
Or just use error code from grep directly as I wrote in the first comment.